pkg/storage/etcd3/compact.go

```diff
@@ -17,13 +17,49 @@ limitations under the License.
 package etcd3
 
 import (
+	"sync"
 	"time"
 
 	"github.com/coreos/etcd/clientv3"
 	"github.com/golang/glog"
 	"golang.org/x/net/context"
 )
 
+const compactInterval = 10 * time.Minute
+
+var (
+	endpointsMap     map[string]struct{}
+	endpointsMapLock sync.Mutex
```

Comment:
move the lock before the map it protects. also rename it to endpointMapMu probably.